sunpy/data/_sample.py

```diff
@@ -51,52 +65,58 @@
     "NORH_LIGHTCURVE": ("tca110810", "")
 }
 
-sample_files = {}
-for key in _files:
-    sample_files[key] = os.path.abspath(os.path.join(sampledata_dir, _files[key][0]))
+
+free_required=145227776
+if disk_usage('/').free > free_required :
```

Comment:
i got it , i think i use this check inside the **download_sample_data** function and then perform a check  .
i could probably make use of break function .